File: app/src/main/java/com/moisesborges/tvaddict/data/ShowsRepository.java
package com.moisesborges.tvaddict.data;

import android.support.annotation.NonNull;

import com.moisesborges.tvaddict.models.Season;
import com.moisesborges.tvaddict.models.Show;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Maybe;
import io.reactivex.Single;

/**
 * Created by Moisés on 12/04/2017.
 */

public interface ShowsRepository {
    Single<List<Show>> getShows();

    Single<Show> getFullShowInfo(int showId);

    Single<Show> saveShow(@NonNull Show show);

    Single<List<Show>> getSavedShows();

    Single<Show> getSavedShow(int showId);

    Single<Show> removeShow(int showId);

    Single<Show> updateShow(Show show);
}

File: app/src/main/java/com/moisesborges/tvaddict/data/ShowsRepository.kt
package com.moisesborges.tvaddict.data

import com.moisesborges.tvaddict.models.Show
import com.moisesborges.tvaddict.models.UpcomingEpisode
import io.reactivex.Single

/**
 * Created by Moisés on 12/04/2017.
 */

interface ShowsRepository {
    fun loadShows(page: Int): Single<List<Show>>

    fun getFullShowInfo(showId: Int): Single<Show>

    fun saveShow(show: Show): Single<Show>

    fun getSavedShows(): Single<List<Show>>

    fun getSavedShow(showId: Int): Single<Show>

    fun removeShow(showId: Int): Single<Show>

    fun updateShow(show: Show): Single<Show>

    fun fetchUpcomingEpisodes(): Single<List<UpcomingEpisode>>

    fun searchShows(showName: String): Single<List<Show>>
}
